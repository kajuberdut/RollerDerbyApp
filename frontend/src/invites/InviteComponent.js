import React, { useState, useEffect } from "react";
import "./InviteComponent.css";
import FastApi from "../Api";
import Loading from "../multiUse/loading/Loading";
  
/**  
 * Display invite component
 */

function InviteComponent({ invite, getInvites, getTeams }) {

    /** Sets sender, team and is loading in state */

    const [ sender, setSender ] = useState();
    const [ team, setTeam ] = useState();
    const [isLoading, setIsLoading] = useState(true);

    /** Retrieve user from local storage*/

    const user = JSON.parse(localStorage.getItem('user'));

    /** API get request for team */

    async function getTeam() {

        try {

          let team = await FastApi.getGroup(invite.teamId);
          setTeam(team);
        } catch (errors) {

          return { success: false, errors };
        }
    }

    /** API get request for sender*/

    async function getSender() {

        try {
            let user = await FastApi.getUserById(invite.senderId);
            setSender(user);
        } catch (errors) {

            return { success: false, errors };
        }
    }

    /** API post request for user group */

    async function acceptInvite() {

        try {

            let data = {
                userId: user.userId,
                groupId: invite.teamId
            }

            await FastApi.addUserToGroup(data);
        } catch (errors) {

            return { success: false, errors };
        }
    }

    /** API put request for team invite*/

    async function changeStatusInvite() {
 
        try {

            let data = {
                status: "accepted"
            }

            await FastApi.updateTeamInvite(invite.inviteId, data);
        } catch (errors) {

            return { success: false, errors };
        }
    }
    
    /** Reloads teams when changes request for teams */

    useEffect(() => {
        getTeam();
        getSender();
        setIsLoading(false);
    }, []);

    
    /** Display loading if API call is has not returned */

    if (isLoading) {
        return (
            <Loading />
        )
    }

    /** Handle click of button  */

    async function handleClick(e) {
        e.preventDefault(); 
        acceptInvite(); 
        changeStatusInvite(); 
        getInvites();
        getTeams();
    }

    /** Render the invite component */
      
    return (
      <div key={"InviteComponent-" + invite.inviteId } className="InviteComponent" style={{width: '77%', minHeight: '90px', height: 'auto', borderRadius: '5px'}}> 
        <div style={{justifyContent: 'flex-end'}}> 
            <div>
                {team && team.name && <h6 style={{paddingTop: '10px', paddingLeft: '10px', paddingRight: '10px'}}>Join Team {team.name}</h6> }
                { sender && <p>Invited By: {sender.username}</p> }
            </div>  
            <div style={{paddingRight: '10px', paddingLeft: '170px', paddingBottom: '10px', margintop: '30px'}}>
                <button onClick={handleClick} style={{height: '30px', borderRadius: '4px'}}>Accept
                </button>    
            </div>     
        </div>
      </div>

    );
}
  
export default InviteComponent
