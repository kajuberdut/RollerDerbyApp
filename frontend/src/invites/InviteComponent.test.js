import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import InviteComponent from "./InviteComponent";
import { BrowserRouter } from "react-router-dom";

// * Passing as of 2/13/24
// npm test InviteComponent.test.js
// must be in frontend directory 


afterEach(() => {
  cleanup()
})


describe("Tests Invite Component Page Renders", () => {

    const invite =  jest.fn().mockResolvedValue(
        [
          { inviteId: 1 },
        ]
      );

    test("invite component should render", () => {
      render(
        <BrowserRouter>
              <InviteComponent invite={invite}/>
        </BrowserRouter>,
      );
    
      expect(screen.getByText('Accept')).toBeInTheDocument();
    }); 
  
    it("matches snapshot", function() {
      const { asFragment } = render(
        <BrowserRouter>
              <InviteComponent invite={invite}/>
        </BrowserRouter>,
      );
        expect(asFragment()).toMatchSnapshot();
    });

})
