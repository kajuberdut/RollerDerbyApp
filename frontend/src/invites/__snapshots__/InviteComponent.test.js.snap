// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Invite Component Page Renders matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="InviteComponent"
    style="width: 77%; min-height: 90px; height: auto; border-radius: 5px;"
  >
    <div
      style="justify-content: flex-end;"
    >
      <div />
      <div
        style="padding-right: 10px; padding-left: 170px; padding-bottom: 10px;"
      >
        <button
          style="height: 30px; border-radius: 4px;"
        >
          Accept
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
